### DISCUSSION TITLE
[Feature]

### DISCUSSION THREAD

**Discussion by ZhangLearning** [open-mmlab/mmcv#2721] [General]
### What is the feature?

你好，我发现了关于 box_iou_rotated.py 中不合理的地方，根据接口描述：
Return the ious betweens boxes. If ``aligned`` is
        ``False``, the shape of ious is (N, M) else (N,).
即返回值在    is (N, M) or (N,)。

因此建议修改 ：
https://github.com/open-mmlab/mmcv/blob/66c5515174110ffab353fa86d9e07a4352c51eed/mmcv/ops/box_iou_rotated.py#L136
为
 ious = bboxes1.new_zeros([rows , cols]) 
并删除 146 和 147 行的 view 操作：
https://github.com/open-mmlab/mmcv/blob/66c5515174110ffab353fa86d9e07a4352c51eed/mmcv/ops/box_iou_rotated.py#L147

根据我前期的测试，可能还需要修改 CPU kernel 的实现，位置在：
https://github.com/open-mmlab/mmcv/blob/66c5515174110ffab353fa86d9e07a4352c51eed/mmcv/ops/csrc/pytorch/cpu/box_iou_rotated.cpp#L24

**Comment by zhouzaida** [comment]
你好，请问你是觉得这样的处理不高效么？

**Reply by jiangjiang-coder66** [reply]
当 aligned 是 False的时候，返回结果会进行 `ious = ious.view(rows, cols)` ，一开始申请二维数组便可以去掉该view操作，或许更好一些。此时需要修改CPU端代码，改成二维数组访问的形式。GPU kernel函数处理的是指针，应该不需要改动。
